temporal-sdk/src/main/java/io/temporal/activity/ActivityInfo.java

```diff
@@ -139,4 +141,8 @@ public interface ActivityInfo {
 
   /** Used to determine if the Activity Execution is a local Activity. */
   boolean isLocal();
+
+  /** Return the priority of the activity task. */
+  @Experimental
+  Priority getPriority();
```

Comment:
What situations is this nullable? Old servers only or always unless explicitly set? (just curious, though if confusing we should consider documenting the cases where it may be)

Also a bit confused on this class how a user can know whether a return is nullable or not. But given the confusion, best to be safe and make it clear in docs. I would consider `Optional<Priority>` if it can be nullable even on newer servers, but meh. Note this same concern does apply as much to user-set values like the activity options.

Reply:
If it can be nullable we should bias towards tagging with `@Nullable` vs optional, it is the preferred approach in Java

Reply:
It can be nullable insofar as you might be using an old server. So, yeah I'll attach the annotation. I believe new servers will always attach the (default) value.

Reply:
Would be worth documenting that and verifying the test server does match the real server here

Reply:
So server leaves this as `null` if unset which is the equivalent of default. In Go, if the server left it as null, we return the default object so I did the same here

Reply:
I think we need to not do that as it will lie. The default values for priorty are not 0, and therefore we might lie and seem like they've been _set to zero_ for a server that actually just doesn't understand priority

Reply:
Zero means default, so zero is not a lie